src/lib/pipeline/DispatchIssueUnit.cc

```diff
@@ -178,8 +178,6 @@ void DispatchIssueUnit::forwardOperands(const span<Register>& registers,
 
   for (size_t i = 0; i < registers.size(); i++) {
     const auto& reg = registers[i];
-    // Flag scoreboard as ready now result is available
-    scoreboard_[reg.type][reg.tag] = true;
```

Comment:
With the introduction of the additional complexity for the in-order core, it's not guaranteed that a uops values will be written to the reg file the cycle after they've been forwarded. Thus, we had to set the register to be ready (the logic of what's been commented on) at WB. Whilst this above logic works for the OoO in its current state and the aforementioned assumption holds, it may not exist in the future so I copied the logic for the OoO core also. It has the same emergent behaviour as the WB unit is ticked first of all the primary pipeline units.

TLDR; it was unsafe and worked on OoO model assumptions which may not hold in the future.